controllers/extendeddaemonsetreplicaset/controller.go

```diff
@@ -163,6 +166,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	reqLogger.V(1).Info("Updating ExtendedDaemonSetReplicaSet status")
 	err = r.updateReplicaSet(replicaSetInstance, newStatus)
 
+	// Garbage collect the failedPodsBackOff map once per minute,
+	// i.e. whenever the seconds [0,60] is less than the reconcile frequency
+	if now.Time.Second() < int(daemonsetInstance.Spec.Strategy.ReconcileFrequency.Duration.Seconds()) {
+		r.failedPodsBackOff.GC()
```

Comment:
could you add a comment about the `GC()` behaviour? 
```suggestion
		// Garbage collect records that have aged past maxDuration (here: 15min).
		r.failedPodsBackOff.GC()
```